File: LeetCode/Practice Problems/sqrtx.cpp
class Solution {
public:
    
    long long binarySearch(int x) {
        int s = 0, e = x;
        long long mid = s + (e - s) / 2;
        long long ans = -1;
        while(s <= e) {
            long long sq = mid * mid;
            
            if(sq == x) {
                return mid;
            }
            else if(sq < x) {
                ans = mid;
                s = mid + 1;
            }
            else {
                e = mid - 1;
            }
            mid = s + (e - s) / 2;
        }
        return ans;
    }
    
    int mySqrt(int x) {
        return binarySearch(x);
    }
};

File: LeetCode/Practice Problems/Easy/sqrtx.java
class Solution {
    public int mySqrt(int x) {
        int s = 1, e = x;
        int m = s + (e - s) / 2;
        
        int ans = 0;
        while(s <= e) {
            if(m <= (x / m)) {
                ans = m;
                s = m + 1;
            }
            else {
                e = m - 1;
            }
            
            m = s + (e - s) / 2;
        }
        
        return ans;
    }
}